webapp/blog/views.py

```diff
@@ -129,6 +135,9 @@ def article(slug):
     if not articles:
         flask.abort(404, "Article not found")
 
+    if day or month or year:
+        return flask.redirect("/blog/{slug}".format(**locals()))
```

Comment:
I try to avoid as much as I can to write urls out of routes functions `@app.route("/my-route")`  because it happened that we had to change our url structure. We had to update every redirect in the code.

To solve this flask gives us the possibility to redirect directly to a function. This avoids to have a strong dependency in the code with the urls. It looks like this:
```python
return flask.redirect(flask.url_for(".article"), slug="{slug}".format(**locals()))
```

Reply:
:+1: